### DISCUSSION TITLE
Using Claude2?

### DISCUSSION THREAD

**Discussion by chen-rn** [machinewrapped/llm-subtrans#51] [General]
Now that both GPT4 and Claude 2 are open for access, I wonder if we can utilize Claude 2's larger context window to do a _better_ job translating.

Have you experimented with either of them yet?

**Comment by machinewrapped** [comment]
I've experimented with GPT4, it seems to produce slightly better translations for a much higher cost (10x) - possibly worth it for some people. It is much better at understanding and summarising context, so it might be a better choice for particularly complicated source material.

I'd like to change to a more generic architecture that can support different endpoints and APIs, it's probably not a _huge_ task to do so. It might be a good opportunity to split the translation kernel off into a separate library from the GUI so that's easier to incorporate in other projects, too.
